src/reflect/scala/reflect/internal/Printers.scala

```diff
@@ -66,7 +68,11 @@ trait Printers extends api.Printers { self: SymbolTable =>
     }
   }
 
-  class TreePrinter(out: PrintWriter) extends super.TreePrinter {
+  @nowarn("msg=shadowing a nested class of a parent is deprecated")
+  @deprecated("use InternalTreePrinter instead", since = "2.13.4")
+  class TreePrinter(out: PrintWriter) extends InternalTreePrinter(out)
+
+  class InternalTreePrinter(out: PrintWriter) extends super.TreePrinter {
```

Comment:
I may have the relationship between these to reversed, and `InternalTreePrinter` ought to extend `this.TreePrinter` for compatibility (so that something accepting a `TreePrinter` still works. I'll have to think on it a bit more

Reply:
I remember making these changes BTW, but I don't remember if it was a PR that didn't make it or what. deja vu!